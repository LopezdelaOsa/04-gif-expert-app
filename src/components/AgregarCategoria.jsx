import { useState } from 'react'

export const AgregarCategoria = ( { onNuevaCategoria } ) => {

    //const [valorInput, setValorInput] = useState('Zelda')
    const [valorInput, setValorInput] = useState('')

    //const onInputCambio = (event) => {
    const onInputCambio = ({ target }) => {
        //console.log(event.target.value)
        //setValorInput(event.target.value)
        setValorInput( target.value )
    }

    const onEnviar = (event) => {
        //console.log(event)
        event.preventDefault()
        //console.log(valorInput)
        if (valorInput.trim().length <= 1) return
        
        //setCategorias (  categorias => [valorInput, ...categorias])
        onNuevaCategoria(valorInput.trim())
        setValorInput('')

    }

  return (
    //<form onSubmit={ (event) => onEnviar(event)}>
    <form onSubmit={ onEnviar }>
        <input
            type="text"
            placeholder="Buscar gifs por tema"
            value={ valorInput }
            //onChange= { (event) => onInputCambio (event) }
            onChange= { onInputCambio  }
        />
    </form>
  )
}
