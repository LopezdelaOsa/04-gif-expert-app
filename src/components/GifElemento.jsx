//import React from 'react'
import PropTypes from 'prop-types'

export const GifElemento = ( { titulo, url, id} ) => {
    //console.log(id, titulo, url)
  return (
    <div className='card'>
        <img src={url} alt={titulo} />
        <p>{titulo}</p>

    </div>
  )
}

GifElemento.propTypes = {
  titulo: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

// Tarea:
// 1. Añadir propTypes
//    titulo obligatorio
//    url obligatorio
// 2. Evaluar snapshoty