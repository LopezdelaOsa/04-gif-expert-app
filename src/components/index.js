// Archivo de barril
export * from './AgregarCategoria'
export * from './GifRejilla'
export * from './GifElemento'