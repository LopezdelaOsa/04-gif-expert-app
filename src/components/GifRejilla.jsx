//import { useEffect, useState } from "react"
//import { pillarGifs } from "../helpers/pillarRejilla"
import { GifElemento } from "./GifElemento"
import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifRejilla = ( { categoria } ) => {

  // const [imagenes, setImagenes] = useState([])

  // const pillarImagenes = async() => {
  //   const nuevasImagenes = await pillarGifs( categoria)
  //   setImagenes(nuevasImagenes)
  // }

  //   useEffect ( () => {
  //     pillarImagenes()
  //   }, [ ]  )

  const { imagenes, isLoading } = useFetchGifs(categoria)

  //console.log({imagenes, isLoading})
  console.log(isLoading)
    
  return (
    <>
        <h3> {categoria} </h3>
        {
          // isLoading ? (<h2>Cargando...</h2>) : null
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className='card-grid'>{
          //imagenes.map( img => (
            //<li key={img.id}>{img.titulo}</li>
          //) )

          //imagenes.map ( ({id, titulo, url}) => (
            //<li key={id}>{titulo}</li>
            //<GifElemento key={ id } titulo={ titulo } url={ url } />
          //))

            imagenes.map( (img) => (
              <GifElemento key={img.id}
                {...img}
              />
            )

            )

        }</div>
 
    </>
  )
}
