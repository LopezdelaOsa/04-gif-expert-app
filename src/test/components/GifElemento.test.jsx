import { render, screen } from "@testing-library/react"
import { GifElemento } from "../../components/GifElemento"

describe('Pruebas en GifElemento', () => {

    const titulo = 'Cualquier titulo vale'
    const url = 'http://caulquier.url.jpg'

    test('Debe coincidir con el snapshot', () => {

        const {container} = render (<GifElemento titulo='{titulo}' url='{url}' />) //container
        expect(container).toMatchSnapshot()

    })

    test('Mostrando la imagen con la URL y el ALT indicadp', () => {
        render (<GifElemento titulo={titulo} url={url} />)
        //screen.debug()
        //expect ( screen.getByRole('img').src).toBe(`${url}/`) 
        //expect ( screen.getByRole('img').alt).toBe(titulo) 
        const { src, alt} = screen.getByRole('img')
        expect(src).toBe(`${url}/`)
        expect(alt).toBe(titulo)

    })

    test('Debe mostrar el titulo como texto en el componente', () => {
        render (<GifElemento titulo={titulo} url={url} />)
        expect ( screen.getByText(titulo) ).toBeTruthy()

    })
})