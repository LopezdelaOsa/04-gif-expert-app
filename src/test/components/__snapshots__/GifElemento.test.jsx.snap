// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifElemento Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="{titulo}"
      src="{url}"
    />
    <p>
      {titulo}
    </p>
  </div>
</div>
`;
